boa/src/builtins/typed_array/integer_indexed_object.rs

```diff
@@ -0,0 +1,118 @@
+use super::TypedArrayName;
+use crate::{
+    gc::{empty_trace, Finalize, Trace},
+    object::{JsObject, ObjectData},
+    Context, JsResult,
+};
+
+/// Type of the array content.
+#[derive(Debug, Clone, Copy, Finalize)]
+pub(super) enum ContentType {
+    Number,
+    BigInt,
+}
+
+unsafe impl Trace for ContentType {
+    // safe beacuase `ContentType` is `Copy`
+    empty_trace!();
+}
+
+/// <https://tc39.es/ecma262/#integer-indexed-exotic-object>
+#[derive(Debug, Default, Clone, Trace, Finalize)]
+pub struct IntegerIndexedObject {
+    pub(super) viewed_array_buffer: Option<JsObject>,
+    pub(super) typed_array_name: Option<TypedArrayName>,
+    pub(super) content_type: Option<ContentType>,
+    pub(super) byte_offset: usize,
+    pub(super) byte_length: usize,
+    pub(super) array_length: usize,
+}
```

Comment:
Is it necessary that `typed_array_name` and `content_type` must be `Option`s?
Also, couldn't we obtain the `ContentType` with a function in `TypedArrayName`?

Reply:
We can indeed get the content type from the typed array name. About the reason for being options, is that they are set up in different points in time. Maybe we could put them when allocating, but not sure for the generic TypedArray. We could also have a Default.